fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -183,8 +193,65 @@ private void prepare(ConnectContext connectContext, OptExpression logicOperatorT
         context.setTraceInfo(new OptimizerTraceInfo(connectContext.getQueryId()));
 
         // process materialized views
-        if (Config.enable_experimental_mv) {
+        // TODO: add session variable
+        if (Config.enable_experimental_mv && !optimizerConfig.isRuleBased()) {
             // TODO: register materialized views
+            // register materialized views
+            registerMaterializedViews(logicOperatorTree, connectContext);
```

Comment:
I think we shouldn't register MV every time.

Reply:
why? every time during optimizer we should prepare the related mvs.